### PR TITLE
[WIP] Better Socket.io instructions

### PR DESCRIPTION
**Pull request by laurencei**
This is an attempt to solve https://github.com/laravel/docs/issues/3173

Essentially there are some concerns that the default example in the docs for Socket.IO has the potential to cause a complete failure of all your JS scripts if the socket.io is not available for any reason (i.e. an outage). So if the server is not running (for any reason) then the JS is not available, and the whole thing falls apart, but brings down the rest of your unrelated JS scripting with it.

The advice from @barryvdh and @kkomelin is to modify the example to make it more robust to failures.

I've put **[WIP]** as I want to get confirmation from @barryvdh and/or @kkomelin that this PR solves their concerns first

### CODE REVIEW

File: broadcasting.md [hunk 1]
```diff
@@ -95,10 +95,16 @@ If you are going to pair the Redis broadcaster with a Socket.IO server, you will
 
     <script src="//{{ Request::getHost() }}:6001/socket.io/socket.io.js"></script>
 
-Next, you will need to instantiate Echo with the `socket.io` connector and a `host`.
+Add the following to your NPM dependancy:
```
Comment (kkomelin) [inline broadcasting.md]: @laurencei "dependency" please

File: broadcasting.md [hunk 2]
```diff
@@ -95,10 +95,16 @@ If you are going to pair the Redis broadcaster with a Socket.IO server, you will
 
     <script src="//{{ Request::getHost() }}:6001/socket.io/socket.io.js"></script>
```
Comment (kkomelin) [inline broadcasting.md]: We don't need this if we add this "window.io = require('socket.io-client');"

**Comment by laurencei** [comment]
@kkomelin - ok, changes made. How is that?

**Comment by taylorotwell** [comment]
Please re-submit when you have personally tested it. Have you?

**Pull request closed without merging**